File: php/bitflip.php
<?php
    $ITERATIONS = 100000;
    $MAXLENGTH = 32768;
    for( $l=16; $l<=$MAXLENGTH; $l*=2 ) {
        $b="";
        for( $i=0; $i<$l; ++$i ) $b.=(rand()/getrandmax()<0.5)?"0":"1";
        $start=microtime(true);
        for( $i=0; $i<$ITERATIONS;++$i) {
            $b[$aleat=rand(0,$l-1)]=($b[$aleat]=="0")?"1":"0";
        }
        $end=microtime(true);
        echo "    PHP-bitflip, ".strlen($b).", ".($end-$start)."\n";
    }
    ?>

File: clojure/bitflip.clj
(ns ga-performance.core
  (:gen-class))

(def iterations 100000)
(def maxlength 32768)

;; bitflip
(loop [len 16]
  (if (<= len maxlength)
    (do
      (let [arr (into [] (map (fn [_] (if (zero? (rand-int 2)) 1 0)) (range len)))]
        (let [start (. System (nanoTime))]
          (loop [k iterations]
            (if (> k 0)
              (do
                (loop [a arr
                       c 0]
                  (if (>= c iterations)
                    true
                    (recur (update-in arr [(rand-int len)] #(bit-flip % 0))
                           (inc c))))
                (recur (dec k)))
              ))
          (println (str "clojure-persistentvector, " len ", " (float (/ (- (. System (nanoTime)) start) 1000000000))))))
      (recur (bit-shift-left len 1)))
    true))
